Source/WebCore/html/HTMLFormElement.cpp

```diff
@@ -748,10 +745,14 @@ bool HTMLFormElement::wasUserSubmitted() const
 
 HTMLFormControlElement* HTMLFormElement::findSubmitter(const Event* event) const
 {
-    if (!event || !is<Node>(event->target()))
+    if (!event)
+        return nullptr;
+    auto* node = dynamicDowncast<Node>(event->target());
+    if (!node)
         return nullptr;
-    auto& node = downcast<Node>(*event->target());
-    auto* element = is<Element>(node) ? &downcast<Element>(node) : node.parentElement();
+    auto* element = dynamicDowncast<Element>(*node);
```

Comment:
Can we dynamicDowncast directly to Element?

Reply:
No, it would break the code. This code needs to be able to distinguish Node vs Element:
1. If not a Node, early return
2. If an Element, use that element
3. If an Node but not an an element, use that node's parentElement.

Reply:
Ah right, I missed the following lines